pkg/gornir/gornir.go

```diff
@@ -44,10 +49,15 @@ func (g *Gornir) RunS(title string, runner Runner, task Task) (chan *JobResult,
 // RunA will execute the task over the hosts in the inventory using the given runner.
 // This function doesn't block, the user can use the method Runnner.Wait instead.
 // It's also up to the user to ennsure the channel is closed
-func (g *Gornir) RunA(title string, runner Runner, task Task, results chan *JobResult) error {
+func (gr *Gornir) RunA(title string, runner Runner, task Task, results chan *JobResult) error {
 	err := runner.Run(
-		NewContext(context.Background(), title, g, g.Logger),
+		context.Background(), // TODO pass this?
```

Comment:
I would pass context.TODO() for now (so is explicit that needs to be re-worked). I can then work on adding a cancellation mechanism to cancel all the child processes if the parent is terminated (to clean up all the goroutines created) . 

We would need to create the context before the Run command and then make sure we terminate the goroutines with the Done channel signal).

```go
ctx := context.Background()
ctx, cancel := context.WithCancel(ctx)
```

On the execution side:

```go
go func() {
	for {
		select {
		case <-ctx.Done():
			return // returning not to leak the goroutine
		...
		}
	}
}()
```

Reply:
I agree